### ISSUE TITLE
Two sessions are created upon registration

### ISSUE THREAD

**Issue by kuba-orlik** [vector-im/riot-web#12865]
### Description

When registering a new account using the web interface, two sessions are created, that causes a "verify this session" popup and confuses users.

### Steps to reproduce

- go to riot web in Firefox
- pick "create account"
- create an account on an instance that supports using email for registration
- receive an email with a registration link
- click the link
- log in
- receive a notification that you need to verify your session. Attempting to verify the session just loops forever and does nothing, causing confusion 
- in settings notice that there are two sessions active, disabling one of them solves the issue (but it's very hard to come up with that!)

I'd expect it to create only one session and not show up the confusing pop-up on first login

Logs being sent: no

![image](https://user-images.githubusercontent.com/2697916/77522200-c007f080-6e7b-11ea-87ff-86d13bbfa8df.png)
![image](https://user-images.githubusercontent.com/2697916/77522217-c5fdd180-6e7b-11ea-92ef-7684e20611cf.png)
![image](https://user-images.githubusercontent.com/2697916/77522221-c8f8c200-6e7b-11ea-9a25-8e12e7c62c0b.png)

### Version information
Version information

*    Platform: web (in-browser)

For the web app:

*    Browser: Firefox 74
*    OS: Manjaro Linux
*    URL: private server, riot-web version: 1.5.13

**Comment by bwindels** [comment]
When trying to reproduce this (with cross-signing enabled) I indeed see two sessions. After clicking the link in the e-mail I hit https://github.com/vector-im/riot-web/issues/10038. I proceeded to login though, and didn't get stuck after logging in on the second tab, I am being asked to go through the complete security step and choose a passphrase. This worked for me and then I got a notification to verify the other session that is indeed confusing.

**Comment by kuba-orlik** [comment]
In my case I don't hit #10038, everything seems normal until the cross-verification popup shows up

**Comment by bwindels** [comment]
@kuba-orlik Interesting, but you do end up with two riot tabs I imagine? Did you have to log in after clicking the e-mail link?

**Comment by kuba-orlik** [comment]
I guess that for at least a small amount of time two tabs are open. I've reproduced this issue on video and kept the other tab to see what happens in it. 

It seems that when I open the link from registration confirmation email, the browser opens it in a new tab and then the previous tab (where I started registration asks for e2e key in the background? Nonetheless I seem to end up with two tabs (in the same browser window), with two sessions. 

I guess that before I just closed the first tab without giving it another thought.

Here's the video with the issue reproduced:

https://vimeo.com/400750961
